main.cpp

```diff
@@ -45,8 +46,48 @@ struct Wrapper
     Wrapper(Type&& t) : val(std::move(t)) 
     { 
         std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
-    }
-};
+    }
+
+    void print()
+    {
+        std::cout << "Wrapper::print(" << val << ")" << std::endl;
+    }
+
+private:
+    Type val;
+};
+
+template<>
+struct Wrapper<Point>
+{
+    Wrapper(Point&& p) : point(std::move(p))
+    {
+        std::cout << "Wrapper(" << typeid(point).name() << ")" << std::endl;
+    }
+
+    void print()
+    {
+        std::cout << "Wrapper::print(" << point.toString() << ")" << std::endl;
+    }
+
+private:
+    Point point;
+};
```

Comment:
Project 9.1 Challenge 1:
There is a lot of repetition between the template class Wrapper, and the template specialization for Wrapper<Point>
Can you figure out how to eliminate the duplicate constructor?
hint: you only have to explicitly specialize that which is unique.